### ISSUE TITLE
BUG: Automatically delete the $symbol in yaml information

### ISSUE THREAD

**Issue by leitaosha** [zsviczian/obsidian-excalidraw-plugin#1484]
**Your environment**
SYSTEM INFO:
	Obsidian version: v1.4.16
	Installer version: v1.4.16
	Operating system: Windows 10 Pro 10.0.22631
	Login status: not logged in
	Insider build toggle: off
	Live preview: on
	Legacy editor: off
	Base theme: dark
	Community theme: none
	Snippets enabled: 0
	Restricted mode: off
	Plugins installed: 1
	Plugins enabled: 1
		1: Excalidraw v2.0.7

RECOMMENDATIONS:
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.

**Describe the bug**

After I tested this plugin individually in the newly created repository I found that it indeed automatically delete $ symbol in yaml information after reload obsidian. I only installed this plug-in in the test library, so it has nothing to do with other plug-ins. This problem occurred in both win10 and win11.

like this:

before: 

```yaml
figureTemplate: $$figureTitle$$ $$i$$$$titleDelim$$$$t$$
tableTemplate: $$tableTitle$$ $$i$$$$titleDelim$$$$t$$
```

after:

```yaml
figureTemplate: figureTitle ititleDelimt
tableTemplate: tableTitle ititleDelimt
```
It is easy to reproduce the bug when restarting obsidian

**To Reproduce**
1. Create a markdown file to enter meta-information like this:

```
---
figureTemplate: $$figureTitle$$ $$i$$$$titleDelim$$$$t$$
tableTemplate: $$tableTitle$$ $$i$$$$titleDelim$$$$t$$
---
```

2. restart obsidian.
4. If it fails, it may require two or three restarts.
5. See error

**Expected behavior**
Because the setting with the $ symbol is required in pandoc-crossref. I don't want $ symbols in meta-information to be deleted

**Screenshots**
before
![图片](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/105116435/df90d2f1-a749-452b-93c5-402054ee2c42)
after
![图片](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/105116435/a594dd3f-be7e-43cc-8a4c-699e200515b5)

**Additional context**

**Comment by zsviczian** [comment]
So this involves non-Excalidraw files?
Sounds very unlikely as Excalidraw does not write any files only Excalidraw files and only when they are open in a view... I will test to see what is going on.

**Comment by leitaosha** [comment]
> So this involves non-Excalidraw files? Sounds very unlikely as Excalidraw does not write any files only Excalidraw files and only when they are open in a view... I will test to see what is going on.

Yes，It's strange. And it will delete $ when enable the Excalidraw. 
![PixPin_2023-12-05_18-12-15](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/105116435/a7404700-4ed4-44b7-9626-014821aae62b)

**Comment by zsviczian** [comment]
I am able to reproduce. This is going to be a very interesting investigation... I have no idea what is happening.

**Comment by zsviczian** [comment]
You don't need to restart obsidian just stop and start excalidraw...

**Comment by leitaosha** [comment]
> You don't need to restart obsidian just stop and start excalidraw...

Yes, I found that. This is an interesting phenomenon that I stumbled upon, since the symbol $ is used in yaml to configure pandoc-crossref. So I hope it will be fixed. I really didn't suspect it was an Excalidraw problem in the first place, after all, the two are irrelevant! 

Thank you for your reply!

**Comment by zsviczian** [comment]
So this is caused by MathJax. Because the version of MathJax in Obsidian does not support SVG export, I load a second instance of MathJax... but to enable MathJax to run in a second instance I am applying a hack renaming the MathJax to ExcalidrawMathJax. The end effect is for some reason when ExcalidrawMathJax is loaded the open document is parsed by Obsidian (assumption on my side) and the parsed result is saved back.

The ExcalidrawMathJax hack was recently introduced to improve Chinese text support. (#1406)
This issue seems to be linked: #1473

This is not good - and clearly a result of my hack... I'll think about a solution.

**Comment by leitaosha** [comment]
It's seems a tricky one. And only the format of $$text$$ in yaml, which deletes the $. But the single $ in yaml and any in main body will not. 
![图片](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/105116435/f25fa6df-9a91-4808-9fe8-3e3b51053d9c)

**Comment by zsviczian** [comment]
Managed to solve this. You can test the solution by 
1) downloading [main.zip](https://github.com/zsviczian/obsidian-excalidraw-plugin/files/13577531/main.zip),
2) unpacking and replacing `vault root\.obsidian\plugins\obsidian-excalidraw-plugin\main.js` with the new file
3) restarting Obsidian

Let me know how your testing goes.

**Comment by zsviczian** [comment]
regarding the $ vs. $$ that is not surprising. the issue is caused by an unexpected behavior of the mathjax package caused by the way I used to load mathjax in the previous version. This is what is fixed in main.zip.

**Comment by leitaosha** [comment]
> Managed to solve this. You can test the solution by设法解决了这个问题。可以通过以下方式测试解决方案
> 
>     1. downloading [main.zip](https://github.com/zsviczian/obsidian-excalidraw-plugin/files/13577531/main.zip), 下载 main.zip，
> 
>     2. unpacking and replacing `vault root\.obsidian\plugins\obsidian-excalidraw-plugin\main.js` with the new file解压缩并替换为 `vault root\.obsidian\plugins\obsidian-excalidraw-plugin\main.js` 新文件
> 
>     3. restarting Obsidian 重新启动黑曜石
> 
> 
> Let me know how your testing goes.让我知道您的测试进展如何。

Great, It worked! Both reload plugin and obsidian.
![PixPin_2023-12-06_16-21-21](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/105116435/e07b767b-0536-40ac-82cc-742c9e1b7e20)

**Comment by zsviczian** [comment]
resolved in 2.0.8

**Comment by leitaosha** [comment]
> resolved in 2.0.8

Thanks a lot！
